### DISCUSSION TITLE
러린이의 입장에서 드는 궁금증,,,

### DISCUSSION THREAD

**Discussion by syeonshin** [huijeong-kim/ministore#7] [Q&A] [answered]
안녕하세요ㅎㅎ
제가 아직 러린이라,, 질문이 웃길수도 있는데,, 

pub struct DataBlock(pub [u8; BLOCK_SIZE]); 

여기서 Data Block은 BlOCK_SIZE 개수 만큼의 u8타입의 값을 가진 array를 Tuple 안에 넣은 것으로 보이는데요 (맞나요..?)
왜 tuple로 만들어진건지 궁금합니당..!

**Comment by syeonshin** [comment]
이것 저것 하찮은 질문이 많이 생길 것 같습니다.. ㅎㅎ 그때마다 Q&A를 주제별로 하나씩 만드는 게 좋을까요? 아니면 이런 류의 내용은 여기에 이어서 질문을 올리는 것이 좋을까요! 의견 부탁드립니다~~!

**Reply by dhihm** [reply]
일단 먼저 첫 번째 질문의 답부터 하자면, 

pub struct DataBlock(pub [u8; BLOCK_SIZE); 는 

pub struct DataBlock {
    pub blocks[u8; BLOCK_SIZE],
}
이것과 동일한 의미인데, 내부 필드에 대한 이름이 있다 없다 차이 뿐 이라고 생각합니다. (이름 지정을 안함)
그렇기 때문에 자신 (DataBlock structure) 내부에 있는 field (array) 접근하기 위해서 아래처럼 **self.0** 을 사용하고 있는 것을 볼 수 있습니다. (첫 번째 필드라는 의미로 0을 사용하는데 튜플의 그것과 동일한 의미 <-- 아래 tuple struct 관련 희정님 코멘트 참고. )

impl Serialize for DataBlock {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: serde::Serializer,
    {
        serializer.serialize_bytes(&self.0)
    }
}

두 번째로 질문 카테고리 관련해서는 일단 지금처럼 편하게 올리고, 질문 들이 많아졌을 경우에 카테고리화 해도 충분 할 것 같습니다!
한 게시물에 꼭 연속해서 올릴 필요는 없고 매번 새 게시물로 작성하시죠! 카테고리는 QnA 하나로 통일.

**Reply by huijeong-kim** [reply, marked as answer]
오!! 저도 한 마디 덧붙이자면 필드로 이름 지정할 가치?? 가 없어 보여 이렇게 했어요. 필드 자체가 스트럭트이기 때문에 굳이...?? DataBlock 안에 data 필드 같은 걸 넣을 필요는 없을 것 같았거든요. 원래는 간단히 type 으로 지정하려다 Serialize/Deserialize 구현을 위해 struct 로 만들었습니다 ㅎㅎ

그리고 이런 걸 tuple struct 라고 부르더라고요! 

https://doc.rust-lang.org/reference/items/structs.html?highlight=Struct#structs

**Reply by huijeong-kim** [reply]
동혁님 두 번째 질문의 답은 왜 지우셨나요 ?.?

**Reply by dhihm** [reply]
엇 왜 취소선이 되었죠? ㅋ 원래 저렇게 쓰는걸 tuple struct라고 하는거였군요.

**Reply by dhihm** [reply]
뭔가 취소선이 먹네요 코멘트에 ㅋㅋ

**Reply by huijeong-kim** [reply]
아하! 얘기하고 취소하신 줄 ㅋㅋㅋ 좋은 것 같아요! 일단은 QnA 에다가 자유롭게 쓰시죠!

**Comment by syeonshin** [comment]
오오 ㅎㅎ 이해했습니다..! 친절한 설명과 답변 감솨합니다 흐흐
